### PR TITLE
Use flow session endpoint

### PR DESCRIPTION
**Pull request by mmclead**
must have this merged from backend before released:

https://github.com/swdotcom/swdc-codemetrics-service/pull/666

### CODE REVIEW

File: src/managers/FlowManager.ts [hunk 1]
```diff
@@ -99,34 +85,17 @@ async function initiateFlow() {
...
-  const status = {
-    status_text: configSettings.slackAwayStatusText,
-    status_emoji: ":large_purple_circle:",
-    status_expiration: 0,
-  };
+  const flowModeSettings = getPreference("flowMode");
 
-  await setSlackStatus(status);
+  // create a FlowSession on backend.  Also handles 3rd party automations (slack, cal, etc)
+  softwarePost("/v1/flow_sessions", { automated: automated }, getItem("jwt"));
```
Comment (xavluiz) [inline src/managers/FlowManager.ts:91]: we can just use these objects like this {automated} as its the same value var for the key var

File: src/managers/FlowManager.ts [hunk 2]
```diff
@@ -99,34 +85,17 @@ async function initiateFlow() {
...
-  // pause slack notifications
-  if (configSettings.pauseSlackNotifications) {
-    await pauseSlackNotifications(false /*showNotification*/, false /*refreshFlowTree*/, true /*isFlowRequest*/);
-  }
-
-  // set to zen mode
+  // update screen mode
   let screenChanged = false;
-  if (configSettings.screenMode.includes("Full Screen")) {
+  const screenMode = flowModeSettings?.editor?.vscode?.screenMode;
```
Comment (xavluiz) [inline src/managers/FlowManager.ts:95]: this is totally fine but I think we should try to make defaults so we don't have to worry about values being available or not, but again, this is fine, thanks.

**Review by xavluiz — APPROVED** [review]
push it! lets start testing :)

File: src/managers/FlowManager.ts (hunk 1 above)
Comment (mmclead) [reply src/managers/FlowManager.ts:91]: I tried that, but my ES list was telling me it wasn't working.   I like that shorthand too.

File: src/managers/FlowManager.ts (hunk 2 above)
Comment (mmclead) [reply src/managers/FlowManager.ts:95]: we do have defaults coming from the backend, so this "should" never be empty.

**Pull request merged**